lib/albinos.integ.tests.cpp

```diff
@@ -0,0 +1,58 @@
+//
+// Created by milerius on 23/11/2019.
+//
+
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include <doctest.h>
+
+#include "Albinos.h"
+#include "Config.hpp"
+
+//! All the tests need the service to be launched
+TEST_CASE("All case")
+{
+    Albinos::Config cfg_to_retrieve = Albinos::Config("None");
+    Albinos::Config albinos_cfg = Albinos::Config("foo");
+    Albinos::Config* ptr = &albinos_cfg;
+    Albinos::Config* retrieve_ptr = &cfg_to_retrieve;
+
+    //! Creation config
+    CHECK_EQ(Albinos::createConfig("toto", &ptr), Albinos::SUCCESS);
+
+    //! Get key
+    Albinos::Key key;
+    CHECK_EQ(Albinos::getConfigKey(ptr, &key), Albinos::SUCCESS);
+    std::string str("key is: [ " + std::string(key.data) + " ]");
+    MESSAGE(str);
+    CHECK(not std::string(key.data).empty());
+    WARN_GT(key.size, 0);
+
+
+    //! Set setting
+    CHECK_EQ(Albinos::setSetting(ptr, "foo", "bar"), Albinos::SUCCESS);
```

Comment:
To avoid potential false positives but mostly confusion, it might be a good idea to change "bar" to something that doesn't have the same character count than "foo".